pkg/build/builder/common.go

```diff
@@ -180,3 +204,23 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface) error {
+	if err := kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
+		// before updating, make sure we are using the latest version of the build
+		latestBuild, err := client.Get(build.Name)
```

Comment:
@csrwng i'm trying to avoid an update conflict to the build object, and i need to get the latest version of the object, the problem is i don't have credentials to do so from the builders, just need a Get()
tried to add this to bootstrap policy, what am i doing wrong? https://github.com/openshift/origin/pull/10817/files#diff-1d38b34f638332aebf97978ff71d9664R128
```
error: An error occured while updating the build status: User "system:serviceaccount:openshift:builder" cannot get builds in project "openshift"
```
cc: @mfojtik  @bparees